### ISSUE TITLE
[ENH] Add Support for Feature Scaling in D2 DataModule Layer.

### ISSUE THREAD

**Issue by PranavBhatP** [jdb78/pytorch-forecasting#1979]
**Is your feature request related to a problem? Please describe.**
The `EncoderDecoderTimeSeriesDataModule` (D2 layer) currently does not apply feature scaling (for continuous variables) as part of its preprocessing pipeline. 

**Describe the solution you'd like**
The scalers can be implemented within the `_preprocess_data` method and called during `setup`. We fit the scalers on the training data and then transform each individual batch inside the `_ProcessedEncoderDecoderDataset`.

**Additional context**
#1974 - Work items for PTFv2.

**Comment by phoeenniixx** [comment]
I thought we decided to add label encoders to D1 layer rather than D2? Or am I missing something here?

**Comment by PranavBhatP** [comment]
@phoeenniixx yes, you're right! That will require a separate issue then, since it is on a different part of the codebase.

**Comment by PranavBhatP** [comment]
Just a question @phoeenniixx . Why would it be better to add label encoding to the D1? I just wanted to recall why we had come to this decision...
